File: OrderManager.Server/Models/Order.cs
namespace OrderManager.Server.Models
{
    //Maps to the Order table in the database
    public class Order
    {
        public int OrderId { get; set; }
        public int CustomerId { get; set; }
        public int SalesPersonId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
    }

    //Used to return nested details about the order on the API
    public class OrderDetails
    {
        public int OrderId { get; set; }
        public int Quantity { get; set; }
        public decimal OrderTotal { get; set; }
        public Customer?  Customer { get; set; }
        public  Employee? SalesPerson { get; set; }
        public Product? Product { get; set; }

        
        //public decimal Price { get; set; } = 0;
        //public decimal OrderTotal => Product.Price * this.Quantity;


    }

}

File: ordermanager.client/src/models/Order.ts


// src/types/Order.ts

import { Customer } from './Customer';
import { Employee } from './Employee';
import { Product } from './Product';

// Matches your C# Order table
export interface Order {
  orderId: number;
  customerId: number;
  salesPersonId: number;
  productId: number;
  quantity: number;
}

// Returned from API with nested details
export interface OrderDetails {
  orderId: number;
  quantity: number;
  orderTotal: number;
  customer?: Customer;
  salesPerson?: Employee;
  product?: Product;
}
